apps/wappush/js/messagedb.js

```diff
@@ -243,7 +278,11 @@ var MessageDB = (function() {
           state.result = message;
```

Comment:
I think this is the easiest solution since it doesn't have any downsides and it doesn't require extra code. Besides if/when WAP Push becomes a component of the SMS app all this thingamagig to delete stuff at the right time because the application is auto-closed will go away.